### ISSUE TITLE
Rule proposal: `no-accessor-recursion`

### ISSUE THREAD

**Issue by Torhal** [sindresorhus/eslint-plugin-unicorn#1052]
Accessing a property inside an accessor with the same name will cause infinite recursion.

## Fail

```js
get foo() {
    return this.foo;
}

set foo(v) {
    this.foo = v;
}
```

## Pass

```js
get foo() {
    return this._foo;
}

set foo(v) {
    this._foo = v;
}

get bar() {
    return that.bar;
}

set bar(v) {
    that.bar = v;
}
```

**Comment by sindresorhus** [comment]
This is now accepted.
